File: Recursion/RightRotating.py


# Check if the word is right rotating word


from collections import deque

w = "pankaj"
wa = "ajpank"

lst = deque()
for i in w:
    lst.append(i)
print("This is lst: ", lst)
lst1 = deque()
for i in wa:
    lst1.append(i)
print("This is lst1: ", lst1)

flag = False
for i in range(len(w)-1, 0, -1):
    lst.appendleft(w[i])
    lst.pop()
    if(lst == lst1):
        flag = True
        break

if(flag):
    print("Yes")
else:
    print("NO")


# if(lst == lst1):
#     print("yes")
# else:
#     print("No")

File: General/RightRotating.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
    string a = "pankaj";
    string b = "ajpank";

    bool flag = true;

    int i;
    for (i = 0; i < b.size(); i++)
    {
        if(a[0] == b[i]){
            break;
        }
    }
    
    int k = 0;
    for (int j = i; j < b.size(); j++,k++)
    {
        cout<<a[k]<<" "<<b[j]<<endl;
        cout<<k<<" "<<j<<endl;

        if(a[k] != b[j]){
            cout<<"This is wrong"<<endl;
            flag = false;
            break;
        }

        if(k == a.size()-1){
            break;
        }

        if(j==b.size()-1){
            j = -1;
        }
    }
    
    if(flag){
        cout<<"Yes It is Right Rotating";
    }else{
        cout<<"No it is not Right Rotating";
    }
    

    return 0;
}


